ziti/edge/types.go

```diff
@@ -170,3 +181,188 @@ func ApiResponseDecode(data interface{}, resp io.Reader) (*ApiResponseMetadata,
 
 	return apiR.Meta, nil
 }
+
+type InteceptDialOptions struct {
+	ConnectTimeoutSeconds *int
+	Identity              *string
+}
+
+type PortRange struct {
+	Low  uint16
+	High uint16
+}
+
+type InterceptV1Config struct {
+	Addresses   []ZitiAddress
+	PortRanges  []*PortRange
+	Protocols   []string
+	SourceIp    *string
+	DialOptions *InteceptDialOptions `json:"dialOptions"`
+	Service     *Service
+}
+
+type DomainName string
+
+func (dn DomainName) Match(hostname string) int {
+	if len(dn) < 3 {
+		return -1
+	}
+
+	if dn[0] == '*' {
```

Comment:
is DomainName trimmed? this could be an annoying bug if whitespace creeps in

Reply:
domain name comes from config that is validated on create